File: frontend/src/types/booking.ts
export interface AvailabilityBlock {
  day_of_week: number; // 0=Sunday, 1=Monday, ..., 6=Saturday
  start_time: string; // HH:MM format
  end_time: string; // HH:MM format
}

export interface BookingType {
  id: string;
  calcom_event_type_id: number | null;
  name: string;
  description: string | null;
  duration_minutes: number;
  location_type: string; // 'video' | 'phone' | 'in_person'
  booking_url: string | null;
  availability: AvailabilityBlock[];
  active: boolean;
  created_at: string;
  updated_at: string;
}

export interface BookingTypeListResponse {
  booking_types: BookingType[];
}

export interface BookingTypeCreate {
  name: string;
  description?: string | null;
  duration_minutes: number;
  location_type: string;
  availability: AvailabilityBlock[];
}

export interface BookingTypeUpdate {
  name?: string;
  description?: string | null;
  duration_minutes?: number;
  location_type?: string;
  availability?: AvailabilityBlock[];
  active?: boolean;
}

// Calendar mode for switching between events view and availability drawing
export type CalendarMode = 'events' | 'availability';

// Pending availability block drawn on calendar (before saving)
export interface PendingAvailabilityBlock {
  id: string;
  start: Date;
  end: Date;
}

File: backend/app/schemas/booking.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class AvailabilityBlock(BaseModel):
    """A single availability time block."""
    day_of_week: int  # 0=Sunday, 1=Monday, ..., 6=Saturday
    start_time: str  # HH:MM format
    end_time: str  # HH:MM format


class BookingTypeBase(BaseModel):
    name: str
    description: Optional[str] = None
    duration_minutes: int = 30
    location_type: str = "video"  # video, phone, in_person


class BookingTypeCreate(BookingTypeBase):
    availability: list[AvailabilityBlock]


class BookingTypeUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    duration_minutes: Optional[int] = None
    location_type: Optional[str] = None
    availability: Optional[list[AvailabilityBlock]] = None
    active: Optional[bool] = None


class BookingTypeResponse(BookingTypeBase):
    id: str
    calcom_event_type_id: Optional[int] = None
    booking_url: Optional[str] = None
    availability: list[AvailabilityBlock] = []
    active: bool = True
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class BookingTypeListResponse(BaseModel):
    booking_types: list[BookingTypeResponse]


class CalcomWebhookPayload(BaseModel):
    """Webhook payload from Cal.com for BOOKING_CREATED event."""
    triggerEvent: str
    createdAt: str
    payload: dict  # Contains booking details
